### PR TITLE
Improvements to dependency doc

### PR DESCRIPTION
**Pull request by jon-turney**
Some reordering to sections in dependency doc, as suggested in https://github.com/mesonbuild/meson/pull/8888#pullrequestreview-686752028, and some slight polishing while I have it open in an editor.

### CODE REVIEW

**Comment by jon-turney** [comment]
Interestingly, looking at the code `dependency('generic', method: 'system')` fails without doing anything, rather than telling you it's nonsense (and likewise for other methods not meaningful for a generic dependency).

File: docs/markdown/Dependencies.md [hunk 1]
```diff
@@ -136,23 +136,49 @@ Since they can be used interchangeably, the rest of the build
...
+and cmake, except when it is part of the base OS image (such as in FreeBSD
+and macOS); OpenGL which has pkg-config on Unices from glvnd or mesa, but has
+no pkg-config on macOS and Windows.
+
+In these cases meson provides convenience wrappers in the form of `system`
+dependencies. Internally these dependencies do exactly what a user would do
+in the build system DSL or with a script, likely calling
+`compiler.find_library()`, setting `link_with` and `include_directories`. By
+putting these in meson upstream the barrier of using them is lowered, as
+projects using meson don't have to re-implement the logic.
```
Comment (jpakkane) [inline docs/markdown/Dependencies.md:171]: "Meson" is a proper name so it should be written properly capitalizes, not as "meson". There are multiple lower case spellings in this section.

Comment (jon-turney) [reply docs/markdown/Dependencies.md]: Done

**Review by eli-schwartz — APPROVED** [review]
Seems a lot cleaner. :)

**Pull request merged**